lib/server/monalisa-synchronization/MonALISASynchronizer.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * @license
+ * Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+ * See http://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+ * All rights not expressly granted are reserved.
+ *
+ * This software is distributed under the terms of the GNU General Public
+ * License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+const { Log } = require('@aliceo2/web-ui');
+const { setupMonALISAInterface } = require('./MonALISAInterface.js');
+const { dataSource } = require('../../database/DataSource.js');
+const { repositories: { DataPassRepository, RunRepository } } = require('../../database/index.js');
+const { extractLhcPeriod } = require('../utilities/extractLhcPeriod.js');
+const { getOrCreateLhcPeriod } = require('../services/lhcPeriod/getOrCreateLhcPeriod.js');
+
+/**
+ * Manger of synchronization with MonALISA
+ */
+class MonALISASynchronizer {
+    /**
+     * Constructor
+     */
+    constructor() {
+        this.logger = new Log(MonALISASynchronizer.name);
+        this.monALISAInterface = null;
+    }
+
+    /**
+     * Synchronize with data from MonALISA
+     * Update Data Passes data in local DB
+     * @return {Promise<void>} promise
+     */
+    async synchronizeDataPassesFromMonALISA() {
+        if (!this.monALISAInterface) {
+            this.monALISAInterface = await setupMonALISAInterface();
+        }
```

Comment:
Please apply IoC and pass the client in the constructor. Here you are making a direct dependency between the synchronizer and the factory, which should not be the case (role of the synchronizer: synchronize. Not create a client)

Reply:
applied